lxd/storage_volumes_utils.go

```diff
@@ -132,6 +123,21 @@ func storagePoolVolumeUsedByGet(s *state.State, requestProjectName string, vol *
 		return []string{}, err
 	}
 
+	// Handle instance volumes.
+	if vol.Type == cluster.StoragePoolVolumeTypeNameContainer || vol.Type == cluster.StoragePoolVolumeTypeNameVM {
+		volName, snapName, isSnap := api.GetParentAndSnapshotName(vol.Name)
+		if isSnap {
+			return []string{api.NewURL().Path(version.APIVersion, "instances", volName, "snapshots", snapName).Project(vol.Project).String()}, nil
+		}
+
+		// If vol is the instance's root volume and it is defined in a profile,
+		// it won't be added to the list by VolumeUsedByInstanceDevices.
+		instancePath := api.NewURL().Path(version.APIVersion, "instances", volName).Project(vol.Project).String()
+		if !slices.Contains(volumeUsedBy, instancePath) {
+			volumeUsedBy = append(volumeUsedBy, instancePath)
+		}
```

Comment:
If the volume is being used in a profile device as an additional drive, instances with this device won't be added to `volumeUsedBy`, right?

This is probably expected anyway, I am asking more for my own understanding.

Reply:
Yes, that's correct; we don't expand devices from profiles here. We want to report everywhere that references exist, not necessarily where the resources are actually attached.

This spot is a special case for root disk devices are defined on a profile. I'll expand the comment.

Reply:
Cool, makes sense to me.
Also good idea on expanding the comment.